crates/libcgroups/src/common.rs

```diff
@@ -316,65 +316,70 @@ pub enum CreateCgroupSetupError {
     Systemd(#[from] systemd::manager::SystemdManagerError),
 }
 
-pub fn create_cgroup_manager<P: Into<PathBuf>>(
-    cgroup_path: P,
-    systemd_cgroup: bool,
-    container_name: &str,
+#[derive(Clone)]
+pub struct CgroupConfig {
+    pub cgroup_path: PathBuf,
+    pub systemd_cgroup: bool,
+    pub container_name: String,
+}
```

Comment:
What's the benefit of this change?

Reply:
This reduces the number of variables I have to store inside container_args. This PR moves the creation of the cgroup manager to the first thing in intermediate process, so we have to store the variables needed to call cgroup create manager. I can either store 3 or store a 1 struct. Also, these variables are easily shown to be clone-able.

Reply:
I didn't say enough. I think making it a structure is a good change. However, I doubt that the type is changing🤔

Reply:
> I doubt that the type is changing

I am not sure I follow what you are saying here? Can you rephrase/elaborate?